tests/components/nest/camera_sdm_test.py

```diff
@@ -0,0 +1,165 @@
+"""
+Test for Nest cameras platform for the Smart Device Management API.
+
+These tests fake out the subscriber/devicemanager, and are not using a real
+pubsub subscriber.
+"""
+
+from google_nest_sdm.auth import AbstractAuth
+from google_nest_sdm.device import Device
+
+from homeassistant.components import camera
+from homeassistant.components.camera import STATE_IDLE
+
+from .common import async_setup_sdm_platform
+
+PLATFORM = "camera"
+CAMERA_DEVICE_TYPE = "sdm.devices.types.CAMERA"
+DEVICE_ID = "some-device-id"
+
+
+class FakeResponse:
+    """A fake web response used for returning results of commands."""
+
+    def __init__(self, json):
+        """Initialize the FakeResponse."""
+        self._json = json
+
+    def raise_for_status(self):
+        """Mimics a successful response status."""
+        pass
+
+    async def json(self):
+        """Return a dict with the response."""
+        return self._json
+
+
+class FakeAuth(AbstractAuth):
+    """Fake authentication object that returns fake responses."""
+
+    def __init__(self, response: FakeResponse):
+        """Initialize the FakeAuth."""
+        super().__init__(None, "")
+        self._response = response
+
+    async def async_get_access_token(self):
+        """Return a fake access token."""
+        return "some-token"
+
+    async def creds(self):
+        """Return a fake creds."""
+        return None
+
+    async def request(self, method: str, url: str, **kwargs):
+        """Pass through the FakeResponse."""
+        return self._response
+
+
+async def async_setup_camera(hass, traits={}, auth=None):
+    """Set up the platform and prerequisites."""
+    devices = {}
+    if traits:
+        devices[DEVICE_ID] = Device.MakeDevice(
+            {
+                "name": DEVICE_ID,
+                "type": CAMERA_DEVICE_TYPE,
+                "traits": traits,
+            },
+            auth=auth,
+        )
+    return await async_setup_sdm_platform(hass, PLATFORM, devices)
+
+
+async def test_no_devices(hass):
+    """Test configuration that returns no devices."""
+    await async_setup_camera(hass)
+    assert len(hass.states.async_all()) == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)